### ISSUE TITLE
Linking XInput

### ISSUE THREAD

**Issue by excaliburHisSheath** [retep998/winapi-rs#77]
I'm trying to get xinput-sys working, and I'm having trouble getting linking setup. The changes I've made can be found [on my fork](https://github.com/excaliburHisSheath/winapi-rs/tree/41a7f8578dabe725d5a6d8d611a548b078a2a51a/lib/xinput-sys), the important files are [Cargo.toml](https://github.com/excaliburHisSheath/winapi-rs/blob/41a7f8578dabe725d5a6d8d611a548b078a2a51a/lib/xinput-sys/Cargo.toml) and [build.rs](https://github.com/excaliburHisSheath/winapi-rs/blob/41a7f8578dabe725d5a6d8d611a548b078a2a51a/lib/xinput-sys/build.rs). When I try to build with the current setup I get the error `ld: cannot find -lxinput`.

I appreciate any help you can offer.

**Comment by retep998** [comment]
Are you using a full MingW-w64 setup or are you relying on the toolchain bundled with Rust? The bundled toolchain only comes with a few libraries and if you need something that isn't there you have to ditch the bundled toolchain and setup a full MingW-w64 instead.

**Comment by excaliburHisSheath** [comment]
Yeah, I'm just using the bundled rust distribution. Would that matter for linking against Windows libraries, though? Does the full MingW distribution do extra stuff to help find the libraries?

**Comment by retep998** [comment]
In order to link to a library, even a Windows system library, you cannot link to a .dll directly, there needs to be a .lib or .a to link to. Thus when using MingW we depend on the .a libraries provided by MingW. Rust bundles the libraries that are necessary for the standard library, but things like xinput are not bundled and must be provided by a full MingW installation.

**Comment by excaliburHisSheath** [comment]
Ah, that's good to know. I'll give it a shot and see if that's where the problem was coming from.

**Comment by chrisxue815** [comment]
I got a similar error `ld: cannot find -ld3d9`, which was later solved by installing the MSVC version of Rust (found [here](http://static.rust-lang.org/dist/index.html)), and running `cargo` from `VS2015 x64 Native Tools Command Prompt`.

**Comment by retep998** [comment]
The current Rust MSVC nightly should actually be able to auto-detect your VS installation so that it can find the libraries without having to use the VS command prompt.

To fix the issue on `-gnu`, the approach I'm taking is to bundle import libraries from mingw so that it will work with Rust out of the box. I just haven't done that for `d3d9` yet.

**Comment by retep998** [comment]
This is no longer an issue as winapi bundles all its own import libraries for MinGW targets.
